source/material_model/rheology/peierls_creep.cc

```diff
@@ -0,0 +1,229 @@
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/material_model/rheology/peierls_creep.h>
+#include <aspect/utilities.h>
+
+#include <deal.II/base/signaling_nan.h>
+#include <deal.II/base/parameter_handler.h>
+
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    namespace Rheology
+    {
+      template <int dim>
+      PeierlsCreep<dim>::PeierlsCreep ()
+      {}
+
+
+
+      template <int dim>
+      double
+      PeierlsCreep<dim>::compute_viscosity (const double strain_rate,
+                                            const double pressure,
+                                            const double temperature,
+                                            const unsigned int composition) const
+      {
+        const unsigned int c = composition;
+
+        double viscosity = 0.0;
+
+        switch (peierls_creep_flow_law)
+          {
+            case viscosity_approx:
+            {
+              /**
+               * An approximation of the Peierls creep formulation, where stress is replaced with strain rate
+               * (second invariant) when calculating viscosity. This substitution requires three additional
+               * fitting parameters (gamma,p,q). Details of this derivation can be found in the following document:
+               * https://ucdavis.app.box.com/s/cl5mwhkjeabol4otrdukfcdwfvg9me4w/file/705438695737
+               * The formulation for the viscosity is:
+               *  ((0.5*gamma*sigma_p)/(A*((gamma*sigma_p)^n)^(1/(s+n)))) * exp(((E+PV)/(R*T))*(((1-gamma^p)^q)/(s+n))) * edot_ii^(1/(s+n)-1)
+               * where
+               * s = ((E + P*V)/(R*T))*p*q*((1 - gamma^p)^(q-1)*(gamma^p)
+               * sigma_p is the Peierls stress
+               * gamma is the Peierls creep fitting parameter
+               * p is the first Peierls creep fitting exponent
+               * q is the second Peierls creep fitting exponent
+               * A is the Peierls prefactor term (1/s 1/Pa^2)
+               * n is the Peierls stress exponent
+               * E is the Peierls activation energy (J/mol)
+               * V is the Peierls activation volume (m^3/mol)
+               * edot_ii is the second invariant of the deviatoric strain rate (1/s)
+               * T is temperature (K)
+               * R is the gas constant
+               */
+              const double s = ( (activation_energies_peierls[c] + pressure * activation_volumes_peierls[c]) / (constants::gas_constant * temperature)) *
```

Comment:
If you remove all `peierls` components from the names below it will be much easier to read ;-). Also maybe there are slightly better names than `left`, `right` and so on. Coarsely it seems `left_term`  is a `stress_term`, `middle_term` is a `arrhenius_term`, and `right_term` is a `strain_rate_term`? Maybe there are better names. Or you could replace them by variable names in the formula above and use the variable names below?

Reply:
I've adopted the naming schemes proposed above, definitely an improvement when reading through 👍 The documentation also now uses the conventions when writing out the formulation.